### PR TITLE
Client Exponential Connection Retry

### PR DESCRIPTION
**Pull request by sancar**
Added exponential retry mechanism when client trying to connect
to cluster.

Cluster connection logic is seperated from client connection manager.
xml and spring configurations are implemented.
tests are added.

### CODE REVIEW

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

File: hazelcast-client/src/main/java/com/hazelcast/client/config/ConnectionRetryConfig.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    /**
+     * by how much to randomize backoffs.
+     * At each iteration calculated back-off is randomized via following method
+     * Random(-jitter * current_backoff, jitter * current_backoff)
+     *
+     * @return jitter
+     */
+    public double setJitter() {
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/config/ConnectionRetryConfig.java]: get?

**Pull request merged**